### ISSUE TITLE
Injected function modify type(XXX).interfaceId

### ISSUE THREAD

**Issue by Amxx** [sc-forks/solidity-coverage#651]
ERC165 enables some form of contract introspection, by advertizing supported interfaces. Some implementation hardcode the interface identifier

```
function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
    return interfaceId == 0x12345678 || super.supportsInterface(interfaceId);
}
```

And others use solidity's built-in interfaceId getter

```
function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
    return interfaceId == type(SomeInterfaceOrContract).interfaceId || super.supportsInterface(interfaceId);
}
```

IMO, these mechanism is to be tested in unit tests, to achieve full coverage of the code. However, we noticed that these tests do fail when `SomeInterfaceOrContract` is an `abstract contract`, possibly because functions are inserted by the coverage tool. This causes the corresponding tests to fail under coverage.

How can we prevent that? The abstract contract I'm talking about does not contain any storage definition, and all function definitions are abstract public or external functions.

**Comment by cgewecke** [comment]
@amxx 

If you have a chance, could you see if the change in PR #660 resolves your test failures?  (It restricts the instrumentation fns to `internal` visibility as long as they are contract (rather than file) scoped).

Have published the branch at tag `eip165` - you can install with: 
```
yarn add solidity-coverage@eip165 --dev
```

**Comment by cgewecke** [comment]
This should be fixed in `0.7.18`. Please just ping if you're still having problems getting these tests to pass.
